Sources/Suggestions/SuggestionProcessing.swift

```diff
@@ -49,21 +58,23 @@ final class SuggestionProcessing {
         }
 
         // Get best matches from history and bookmarks
-        let allLocalSuggestions = localSuggestions(from: history, bookmarks: bookmarks, internalPages: internalPages, query: query)
+        let allLocalSuggestions = localSuggestions(from: history, bookmarks: bookmarks, internalPages: internalPages, openTabs: openTabs, query: query)
 
         // Combine HaB and domains into navigational suggestions and remove duplicates
         let navigationalSuggestions = allLocalSuggestions + duckDuckGoDomainSuggestions
 
         let maximumOfNavigationalSuggestions = min(
             Self.maximumNumberOfSuggestions - Self.minimumNumberInSuggestionGroup,
             query.count + 1)
-        let mergedSuggestions = merge(navigationalSuggestions, maximum: maximumOfNavigationalSuggestions)
+        let expandedSuggestions = replaceHistoryWithBookmarksAndTabs(navigationalSuggestions)
+
+        let dedupedNavigationalSuggestions = Array(dedupSuggestions(expandedSuggestions).prefix(maximumOfNavigationalSuggestions))
```

Comment:
Should behave the same as before but split out to make reasoning a bit easier.  Effectively history items are replaced by the first matching bookmark and/or tab that is found, but otherwise the list is not changed.  Then we run a deduplication across the whole list.